File: jcv-core/src/main/java/com/ekino/oss/jcv/core/comparator/NotNullComparator.java
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator;

import java.util.Objects;

import com.ekino.oss.jcv.core.JsonValueComparator;

import org.skyscreamer.jsonassert.ValueMatcherException;

public class NotNullComparator implements JsonValueComparator<Object> {

    @Override
    public boolean hasCorrectValue(Object actual, Object expected) {
        if (actual != null && !(actual instanceof String) && !"null".equals(actual.toString())) {
            return true;
        }
        throw new ValueMatcherException("Value should not be null", Objects.toString(expected), Objects.toString(actual));
    }
}

File: jcv-core/src/main/kotlin/com/ekino/oss/jcv/core/comparator/NotNullComparator.kt
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator

import com.ekino.oss.jcv.core.JsonValueComparator
import org.skyscreamer.jsonassert.ValueMatcherException
import java.util.Objects

class NotNullComparator : JsonValueComparator<Any> {

  override fun hasCorrectValue(actual: Any?, expected: Any?): Boolean {
    if (actual != null && actual !is String && "null" != actual.toString()) {
      return true
    }
    throw ValueMatcherException("Value should not be null", Objects.toString(expected), Objects.toString(actual))
  }
}
